issuetracking/github.go

```diff
@@ -108,54 +109,87 @@ func (gh GitHub) AssignIssue(issue common.Issue, login string) {
 	}
 }
 
-func (gh GitHub) AddIssueLabel(issue common.Issue, labelName string) error {
-	repo := versioncontrol.Git{}.Repo()
+func (gh GitHub) AddIssueLabel(issue common.Issue, labelName string) {
+	if labelName == "" {
+		log.Println("Can not apply blank label.")
+		return
+	}
 
-	client := gh.client()
+	if issue.HasLabel(labelName) {
+		msg := fmt.Sprintf("Issue %d already has label '%s'.", *issue.Number, labelName)
+		log.Println(msg)
+		return
+	}
 
 	_, err := gh.getLabel(labelName)
 
 	if err != nil {
-		return errors.New(fmt.Sprintf("Label '%v' does not exist.", labelName))
+		msg := fmt.Sprintf("Unable to find label '%s'. Please make sure it exists.", labelName)
+		log.Println(msg)
+		return
 	}
 
+	repo := versioncontrol.Git{}.Repo()
 	labels := []string{labelName}
 
-	_, _, err = client.Issues.AddLabelsToIssue(
+	_, _, err = gh.client().Issues.AddLabelsToIssue(
 		context.Background(),
 		repo.Owner,
 		repo.Name,
 		*issue.Number,
 		labels,
 	)
 
+	msg := ""
+
 	if err != nil {
-		return err
+		msg = fmt.Sprintf("Failed to add label '%s' to issue %d.", labelName, *issue.Number)
+	} else {
+		msg = fmt.Sprintf("Added label '%s' to issue %d.", labelName, *issue.Number)
 	}
 
-	fmt.Println("Added label '%s' to issue %d", labelName, issue.Number)
-
-	return nil
+	log.Println(msg)
 }
 
 func (gh GitHub) RemoveIssueLabel(issue common.Issue, labelName string) {
-	repo := versioncontrol.Git{}.Repo()
+	if labelName == "" {
+		log.Println("Can not apply blank label.")
+		return
+	}
 
-	client := gh.client()
+	if !issue.HasLabel(labelName) {
+		msg := fmt.Sprintf("Issue %d does not have label '%s'.", *issue.Number, labelName)
+		log.Println(msg)
+		return
+	}
+	
+	_, err := gh.getLabel(labelName)
 
-	_, err := client.Issues.RemoveLabelForIssue(
+	if err != nil {
+		msg := fmt.Sprintf("Unable to find label '%s'. Please make sure it exists.", labelName)
+		log.Println(msg)
+		return
+	}
+
+	repo := versioncontrol.Git{}.Repo()
+	
+	_, err = gh.client().Issues.RemoveLabelForIssue(
 		context.Background(),
 		repo.Owner,
 		repo.Name,
 		*issue.Number,
 		labelName,
 	)
 
+	msg := ""
+
 	if err != nil {
-		panic(err)
+		msg = fmt.Sprintf("Failed to remove label '%s' from issue %d.", labelName, *issue.Number)
```

Comment:
You mean showing the actual GH error that comes back?

Reply:
Exactly :)
Showing our own message is fine but without seeing what the server returns it will make it hard to figure out if it's because the label isn't there or if there was a real problem.